src/raftstore/store/util.rs

```diff
@@ -323,12 +323,30 @@ pub fn get_region_approximate_size_cf(
     let (_, mut size) = db.get_approximate_memtable_stats_cf(cf, &range);
     let collection = db.get_properties_of_tables_in_range(cf, &[range])?;
     for (_, v) in &*collection {
-        let props = SizeProperties::decode(v.user_collected_properties())?;
+        let props = RangeProperties::decode(v.user_collected_properties())?;
         size += props.get_approximate_size_in_range(&start, &end);
     }
     Ok(size)
 }
 
+pub fn get_region_approximate_keys_cf(
+    db: &DB,
+    cfname: &str,
+    region: &metapb::Region,
+) -> Result<u64> {
+    let cf = rocksdb_util::get_cf_handle(db, cfname)?;
+    let start = keys::enc_start_key(region);
+    let end = keys::enc_end_key(region);
+    let range = Range::new(&start, &end);
+    let (_, mut size) = db.get_approximate_memtable_stats_cf(cf, &range);
+    let collection = db.get_properties_of_tables_in_range(cf, &[range])?;
+    for (_, v) in &*collection {
+        let props = RangeProperties::decode(v.user_collected_properties())?;
+        size += props.get_approximate_keys_in_range(&start, &end);
```

Comment:
When we first upgrade, this will be mostly zero, you need to check if it works in that case too.

Reply:
Here a smaller `keys` will not cause any change, only a bigger one may cause the split operation.
And only both `keys` and `size` are small enough will cause a merge operation（while first upgrade, the `keys` will mostly smaller than the actual value, then the merge condition check will be based on `size` which works like before）